src/autowiring/AutoPacket.cpp

```diff
@@ -274,6 +271,48 @@ void AutoPacket::UpdateSatisfactionUnsafe(std::unique_lock<std::mutex> lk, const
 
   // Recursively mark unsatisfiable any single-output arguments on these subscribers:
   std::vector<const AutoFilterArgument*> unsatOutputArgs;
+  auto MarkOutputsUnsat = [&unsatOutputArgs] (const SatCounter& satCounter) {
+    const auto* args = satCounter.GetAutoFilterArguments();
+    for (size_t i = satCounter.GetArity(); i--;) {
+      // Only consider output arguments:
+      if (args[i].is_output)
+        // This output is transitively unsatisfiable, include it for later removal
+        unsatOutputArgs.push_back(&args[i]);
+    }
+  };
+
+  for (auto modifier : disposition.m_modifiers) {
+    if (!modifier.satCounter)
+      continue;
+    auto& satCounter = *modifier.satCounter;
+    if (modifier.is_shared) {
+      if (disposition.m_decorations.size() > 1)
+        throw autowiring_error("An AutoFilter was detected which has single-decorate rvalue shared pointer argument in a graph with multi-decorate outputs");
```

Comment:
The tricky thing here is because shared pointer modifier can do things like `sp.reset()`, `sp.reset(new Decoration<1>)`. It can change the number of decorations (`disposition.m_decorations.size()`).
That is why when we found out an modifier `is_shared`, we call it right there (along with everything else already in the `callQueue`) rather than put it in the `callQueue`. And then check `disposition.m_decorations.size()` inside the loop, because it may change after the call to shared pointer modifier.